integration_tests/testsuite/test_custom.py

```diff
@@ -39,19 +57,179 @@ def runTest(self):
         if not output:
             logging.debug('No custom tests specified.')
         else:
-            for test_info in json.loads(output):
+            for specification in json.loads(output):
                 test_num += 1
-                name = test_info.get('name', 'test_{0}'.format(test_num))
-                path = test_info.get('path')
-                if path is None:
-                    logging.warn('Test \'%s\' has no path specified! '
-                                 'Skipping...', name)
-                    continue
+                self._run_test_for_specification(specification, test_num)
+
+    def _run_test_for_specification(self, specification, test_num):
+        """Given the specification for a test execute the steps and
+            validate the result.
+
+        Args:
+            specification: Dictionary containing the specification.
+            test_num: Identifier of the test.
+
+        Returns:
+            None.
+        """
+        name = specification.get('name', 'test_{0}'.format(test_num))
+        timeout = specification.get('timeout', 3000)
+        steps = specification.get('steps', [])
+        validation = specification.get('validation')
+
+        logging.info('Running custom test: %s', name)
+
+        if self._test_for_old_specification(specification):
```

Comment:
`old` is really not a good name, it doesn't describe the behaviour of the test.
Let's rename it to `simple_test` and `composite_test`.